src/dev/lupio/lupio_pic.cc

```diff
@@ -48,6 +49,11 @@ LupioPIC::LupioPIC(const Params &params) :
     nThread(params.num_threads),
     intType(params.int_type)
 {
+    memset(enable, 0, LUPIO_PIC_NCPU * sizeof(uint32_t));
+    memset(mask, 0, LUPIO_PIC_NCPU * sizeof(uint32_t));
```

Comment:
I'm pretty sure you can do the following, which is more idiomatic C++:

```suggestion
    enable = {0};
    mask = {0};
```

This is also better in the case that you ever change the type of `enable` or `mask` to `uint64_t`

Reply:
I know that `{0}` works when it's within the same statement as the variable declaration itself (e.g., `int enable[10] = {0};`) but are you sure that it also works after the variable has been declared. Otherwise, another possibility would be to do the init as part of the initializer list? (disclaimer, I'm not a C++ guru)

Reply:
Ah, 0 is special. See https://en.cppreference.com/w/c/language/initialization#Zero_initialization

You can also write `enable = {}` to zero initialize as well.